### ISSUE TITLE
`@template` documents declared on extension aren't shown elsewhere

### ISSUE THREAD

**Issue by dickermoshe** [Dart-Code/Dart-Code#5284]
**Describe the bug**

**To Reproduce**
1. Declare a template doccomment on a method in an extension
2. Use elsewhere
3. Hover over method

```
```dart
class Foo {
  /// {@macro baz}
  ///
  /// {@macro pink}
  ///
  ///
  ///
  ///
  ///
  ///
  ///
  void bar() {}

  /// {@template pink}
  /// this works
  /// {@endtemplate}
  void pink() {}
}

extension on Foo {
  /// {@template baz}
  /// This does not work
  /// {@endtemplate}
  void baz() {}
}
```

**Expected behavior**
See the referenced template.
The template is shown if running `dart doc`, so this isn't an issue there.

**Screenshots**

As you can see, the `baz` template isn't rendered in the hover.

<img width="442" alt="image" src="https://github.com/user-attachments/assets/284d1c54-894c-465d-aa4c-11fc0713264c">

<details>

<summary><strong>Workspace Environment</strong></summary>

```text
Dart Code extension: 3.96.0
Flutter extension: 3.96.0 (not activated)

App: Visual Studio Code
App Host: desktop
Remote: wsl
Host Kind: wsl
Version: linux 1.93.1

Workspace type: Dart (LSP)
Workspace name: hi [WSL: Ubuntu]

Dart (3.5.3): /home/mick/development/flutter/bin/cache/dart-sdk
Flutter (3.24.3): /home/mick/development/flutter (No device)

```

</details>

<details>

<summary><strong>Output from 'dart info'</strong></summary>

`/home/mick/development/flutter/bin/cache/dart-sdk/bin/dart info`

If providing this information as part of reporting a bug, please review the information
below to ensure it only contains things you're comfortable posting publicly.

#### General info

- Dart 3.5.3 (stable) (Wed Sep 11 16:22:47 2024 +0000) on "linux_x64"
- on linux / Linux 5.15.153.1-microsoft-standard-WSL2 #1 SMP Fri Mar 29 23:14:13 UTC 2024
- locale is C.UTF-8

#### Project info

- sdk constraint: '^3.5.3'
- dependencies: 
- dev_dependencies: lints, test

#### Process info

| Memory |  CPU | Elapsed time | Command line                                                                           |
| -----: | ---: | -----------: | -------------------------------------------------------------------------------------- |
|  72 MB | 0.0% |        11:11 | dart devtools --machine --allow-embedding --dtd-uri ws:<path>/4yC8DSkXExpGaiRX         |
| 951 MB | 1.7% |        11:11 | dart language-server --protocol=lsp --client-id=VS-Code-Remote --client-version=3.96.0 |
| 544 MB | 1.7% |        10:57 | dart language-server --protocol=lsp --client-id=VS-Code-Remote --client-version=3.96.0 |
|  81 MB | 0.0% |        11:11 | dart tooling-daemon --machine                                                          |
|  82 MB | 0.0% |        10:57 | dart tooling-daemon --machine                                                          |
|  90 MB | 0.1% |        11:11 | flutter_tools.snapshot daemon --show-web-server-device                                 |

</details>

**Comment by DanTup** [comment]
Fixed by https://github.com/dart-lang/sdk/commit/1833e7665801a0c153a9d88710d29b8bdf879c14. The change is in the Dart analysis server, so it'll show up in a future Dart SDK release (not a VS Code extension release).

**Comment by dickermoshe** [comment]
From a flutter standpoint, this won't be fixed until 3.27, right?
